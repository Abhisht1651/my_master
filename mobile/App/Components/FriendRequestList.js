// @flow
import React from 'react'
import PropTypes from 'prop-types'
import { View, Text, TouchableOpacity, Image } from 'react-native'
import styles from './Styles/FriendRequestListStyle'
import { X3SButton } from './Buttons/X3SButton'
import Images from '../Themes/Images'
import MediumProfilePicture from './ProfilePicture/MediumProfilePictures/MediumProfilePicture'

type Props = {
  item: {
    name: string,
    optionalText: string,
    daysText: string,
    image: Image,
    online: boolean,
    id: string
  },
  onAcceptPress: PropTypes.func,
  onIgnorePress: PropTypes.func
}

const FriendRequestList = (props: Props) => {
  const { item, onAcceptPress, onIgnorePress } = props
  return (
    <View style={item.online ? styles.containerwhite : styles.container}>
      <View style={item.online ? styles.dot : styles.dotwhite} />
      <TouchableOpacity style={styles.imageContainer} activeOpacity={1}>
        <MediumProfilePicture image={item.image} />
      </TouchableOpacity>
      <View style={styles.textBlock}>
        <Text style={styles.nameText}>{item.name}</Text>
        <Text style={styles.optionalText}>{item.optionalText}</Text>
        <View style={styles.btnContainer}>
          <View style={styles.button}>
            <X3SButton style={styles.buttonStyle} title="Accept" onPress={onAcceptPress} />
          </View>
          <View style={styles.button}>
            <X3SButton style={styles.buttonStyle} outline title="Ignore" onPress={onIgnorePress} />
          </View>
        </View>
      </View>
      <View style={styles.daysTextContainer}>
        <Text style={styles.daysText}>{item.daysText}</Text>
      </View>
    </View>
  )
}

FriendRequestList.defaultProps = {
  item: {
    name: 'Armando Santanera',
    optionalText: '48 Friends in Common',
    daysText: '1 day ago',
    image: Images.profilePicture,
    online: true,
    id: ''
  },
  onAcceptPress: () => {},
  onIgnorePress: () => {}
}

export default FriendRequestList
