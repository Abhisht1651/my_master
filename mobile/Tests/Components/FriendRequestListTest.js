import { shallow } from 'enzyme'
import React from 'react'

import FriendRequestList from '../../App/Components/FriendRequestList'
import { X3SButton } from '../../App/Components/Buttons/X3SButton'

describe('FriendRequestList', () => {
  it('should match to snapshot', () => {
    const component = shallow(<FriendRequestList />)
    expect(component).toMatchSnapshot()
  })

  it('should render Text ', () => {
    const wrapper = shallow(<FriendRequestList />)
    expect(wrapper.find('Text')).toHaveLength(3)
  })

  it('should render MediumProfilePicture ', () => {
    const wrapper = shallow(<FriendRequestList />)
    expect(wrapper.find('MediumProfilePicture')).toHaveLength(1)
  })
  it('should render X3SButton ', () => {
    const wrapper = shallow(<FriendRequestList />)
    expect(wrapper.find(X3SButton)).toHaveLength(2)
  })
})
