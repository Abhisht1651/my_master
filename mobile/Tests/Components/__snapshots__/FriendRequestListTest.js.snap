// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FriendRequestList should match to snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "borderBottomWidth": 1,
      "borderColor": "#E6ECED",
      "flexDirection": "row",
      "height": 213.57142857142856,
      "paddingTop": 32.857142857142854,
      "width": 750,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#7B46E4",
        "borderRadius": 8,
        "height": 16,
        "justifyContent": "center",
        "marginLeft": 24,
        "marginRight": 16,
        "marginTop": 24.642857142857142,
        "width": 16,
      }
    }
  />
  <ForwardRef
    activeOpacity={1}
    style={
      Object {
        "alignItems": "center",
        "aspectRatio": 1,
        "borderColor": "#DCDCE9",
        "borderRadius": 50,
        "borderWidth": 1,
        "height": 73.28571428571428,
        "justifyContent": "center",
        "marginRight": 26,
      }
    }
  >
    <MediumProfilePicture
      image={
        Object {
          "default": "",
        }
      }
      onPress={[Function]}
      type={1}
    />
  </ForwardRef>
  <View
    style={
      Object {
        "alignItems": "flex-start",
        "width": "50%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#1F1F60",
          "fontFamily": "BananaGrotesk-Medium",
          "fontSize": 30,
          "letterSpacing": -0.12,
        }
      }
    >
      Armando Santanera
    </Text>
    <Text
      style={
        Object {
          "color": "#8F8FAF",
          "fontFamily": "BananaGrotesk-Regular",
          "fontSize": 23,
          "letterSpacing": -0.08,
        }
      }
    >
      48 Friends in Common
    </Text>
    <View
      style={
        Object {
          "flexDirection": "row",
          "marginBottom": 26.285714285714285,
          "marginTop": 24.642857142857142,
        }
      }
    >
      <View
        style={
          Object {
            "marginRight": 20,
          }
        }
      >
        <X3SButton
          onPress={[Function]}
          outline={false}
          style={
            Object {
              "textAlign": "center",
              "width": 150,
            }
          }
          title="Accept"
        />
      </View>
      <View
        style={
          Object {
            "marginRight": 20,
          }
        }
      >
        <X3SButton
          onPress={[Function]}
          outline={true}
          style={
            Object {
              "textAlign": "center",
              "width": 150,
            }
          }
          title="Ignore"
        />
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "flex-end",
        "marginRight": 52,
        "marginTop": "1%",
        "width": "23%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#8F8FAF",
          "fontFamily": "BananaGrotesk-Regular",
          "fontSize": 20,
          "letterSpacing": -0.07,
        }
      }
    >
      1 day ago
    </Text>
  </View>
</View>
`;
